File: NeuroGest.API/Models/Funcionario.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NeuroGest.API.Models;

[Table("funcionarios")]
public class Funcionario
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    [Column("nome")]
    public string Nome { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    [Column("email")]
    public string Email { get; set; } = string.Empty;

    [Required]
    [Column("senha_hash")]
    public string SenhaHash { get; set; } = string.Empty;

    // Ex: "Terapeuta", "Recepcionista", "Coordenador"
    [Required]
    [MaxLength(100)]
    [Column("funcao")]
    public string Funcao { get; set; } = string.Empty;

    // "admin" | "gerente" | "usuario"
    [Required]
    [MaxLength(20)]
    [Column("perfil")]
    public string Perfil { get; set; } = "usuario";

    [MaxLength(20)]
    [Column("telefone")]
    public string? Telefone { get; set; }

    [Column("ativo")]
    public bool Ativo { get; set; } = true;

    [Column("criado_em")]
    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;
}

File: frontend/lib/models/funcionario.dart
class Funcionario {
  final int id;
  final String nome;
  final String email;
  final String funcao;
  final String perfil;
  final String? telefone;
  final bool ativo;
  final DateTime criadoEm;

  Funcionario({
    required this.id,
    required this.nome,
    required this.email,
    required this.funcao,
    required this.perfil,
    this.telefone,
    required this.ativo,
    required this.criadoEm,
  });

  factory Funcionario.fromJson(Map<String, dynamic> json) {
    return Funcionario(
      id:       json['id'],
      nome:     json['nome'],
      email:    json['email'],
      funcao:   json['funcao'],
      perfil:   json['perfil'],
      telefone: json['telefone'],
      ativo:    json['ativo'],
      criadoEm: DateTime.parse(json['criadoEm']),
    );
  }
}
